File: 1004-max-consecutive-ones-iii/1004-max-consecutive-ones-iii.java
class Solution {
    public int longestOnes(int[] nums, int k) {
        int zeroes = 0, max = 0, start = 0;
        for (int end = 0; end < nums.length; end++) {
            if (nums[end] == 0) zeroes++;
            while (zeroes > k) {
                if (nums[start++] == 0) zeroes--;
            }
            max = Math.max(end - start + 1, max);
        }
        return max;
    }
}

File: 1004-max-consecutive-ones-iii/1004-max-consecutive-ones-iii.kt
class Solution {
    fun longestOnes(nums: IntArray, k: Int): Int {
        var (max, zeroes, start) = listOf(0, 0, 0)
        for ((end, e) in nums.withIndex()) {
            if (e == 0) zeroes++
            while (zeroes > k) if (nums[start++] == 0) zeroes--
            max = maxOf(max, end - start + 1)
        }
        return max
    }
}